Formula/opencascade.rb

```diff
@@ -0,0 +1,73 @@
+class Opencascade < Formula
+  desc "3D modeling and numerical simulation software for CAD/CAM/CAE"
+  homepage "https://dev.opencascade.org/"
+  url "https://github.com/FreeCAD/homebrew-freecad/releases/download/0/opencascade-7.2.0.tgz"
+  sha256 "f9a81ca2267635773af9fff80625735dd4f65389af986fe93a94263081b422e9"
+
+  option "with-extras", "Install documentation (~17 MB), source files (~113 MB), samples and templates"
+
+  depends_on "cmake" => :build
+  depends_on "freetype"
+  depends_on "freeimage"
+  depends_on "tbb"
+  depends_on "gl2ps"
+  depends_on "doxygen" if build.with? "extras"
+
+  def install
+    cmake_args = std_cmake_args
+    sdk = MacOS::CLT.installed? ? "" : MacOS.sdk_path
+    opencl_path = Pathname.new "#{sdk}/System/Library/Frameworks/OpenCL.framework/Versions/Current"
+    cmake_args << "-DCMAKE_PREFIX_PATH:PATH=#{HOMEBREW_PREFIX}"
+    cmake_args << "-DCMAKE_INCLUDE_PATH:PATH=#{HOMEBREW_PREFIX}/lib"
+    cmake_args << "-DCMAKE_FRAMEWORK_PATH:PATH=#{HOMEBREW_PREFIX}/Frameworks"
+    cmake_args << "-D3RDPARTY_DIR:PATH=#{HOMEBREW_PREFIX}"
+    cmake_args << "-DUSE_OPENCL:BOOL=ON"
+    cmake_args << "-DUSE_GL2PS:BOOL=ON"
+    cmake_args << "-DUSE_FREEIMAGE:BOOL=ON"
+    cmake_args << "-DUSE_TBB:BOOL=ON"
+    cmake_args << "-D3RDPARTY_TBB_DIR:PATH=#{HOMEBREW_PREFIX}"
+    cmake_args << "-D3RDPARTY_TCL_DIR:PATH=/usr"
```

Comment:
Wondering if this shouldn't be `#{sdk}/usr`, given the lines below

Reply:
The legacy formula uses the macOS built-in tcl-tk so this path is correct.  

However, in the context of this comment, is there a Homebrew best-practice or standard for determining the precedence of using macOS built-in dependencies vs. Homebrew dependencies when developing formula?  For example, should formulae always depend upon Homebrew packages if a dependency is available in Homebrew and fallback to macOS packages or vice-versa?  What are your thoughts on using macOS tcl-tk or the homebrew tcl-tk?